### DISCUSSION TITLE
How do I get a WSL compatible tar file out of a existing flake-enabled NixOS configuration?

### DISCUSSION THREAD

**Discussion by volkswagenfeature** [nix-community/NixOS-WSL#195] [Q&A]
So I'm trying to port over my configs from my laptop to a WSL environment. You can see my up-to-date laptop config [here](https://github.com/volkswagenfeature/nix-sync/tree/BetaBlue-NixOS-2022) or my attempt to make WSL work [here](https://github.com/volkswagenfeature/nix-sync/tree/NixOS_WSL). I'm still learning Nix, so I'm not sure how to call your code from my own. 

If you'll pardon the broad question, What modifcations need to be made to a `flake.nix` to let me produce a tarball of the whole system, with all the stuff I added included? I did get a successful build by adding NixOS-WSL to my flake inputs, and adding the line `defaultPackage."${system}" = inputs.NixOS-WSL.nixosConfigurations.mysystem.config.system.build.installer;` but as you might have expected, this only spits back the "empty", default environment.

**Comment by nzbr** [comment]
You can take a look at [configuration.nix](https://github.com/nix-community/NixOS-WSL/blob/main/configuration.nix) and copy the `wsl` block from there. Then, just add `inputs.NixOS-WSL.nixosModules.wsl` to your configuration's imports. After that, you should be able to build the `config.system.build.installer` of the configuration
